folsom-opencensus/src/main/java/com/spotify/folsom/opencensus/OpenCensusTracer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2019 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.spotify.folsom.opencensus;
+
+import static io.opencensus.trace.AttributeValue.stringAttributeValue;
+import static io.opencensus.trace.Span.Kind.CLIENT;
+import static java.util.Objects.requireNonNull;
+
+import com.spotify.folsom.Span;
+import com.spotify.folsom.Tracer;
+import io.opencensus.trace.Tracing;
+import java.util.concurrent.CompletionStage;
+
+class OpenCensusTracer implements Tracer {
+
+  @Override
+  public Span span(final String name, final CompletionStage<?> future, final String operation) {
+    return internalSpan(name, future, operation, null);
+  }
+
+  @Override
+  public Span span(
+      final String name,
+      final CompletionStage<?> future,
+      final String operation,
+      final String key) {
+    requireNonNull(key);
+    return internalSpan(name, future, operation, key);
+  }
+
+  @SuppressWarnings("MustBeClosedChecker")
+  private Span internalSpan(
+      final String name,
+      final CompletionStage<?> future,
+      final String operation,
+      final String key) {
+    requireNonNull(name);
+    requireNonNull(future);
+    requireNonNull(operation);
+
+    final io.opencensus.trace.Tracer tracer = Tracing.getTracer();
+
+    final io.opencensus.trace.Span ocSpan =
+        tracer.spanBuilder(name).setSpanKind(CLIENT).startSpan();
+
+    ocSpan.putAttribute("component", stringAttributeValue("folsom"));
+    ocSpan.putAttribute("peer.service", stringAttributeValue("memcache"));
+    ocSpan.putAttribute("operation", stringAttributeValue(operation));
+    if (key != null) {
```

Comment:
should adding the key as an annotation be configurable? I'm thinking there might be scenarios where key is "sensitive."

Reply:
I was thinking about this, which is why I opted not to include the value. But perhaps this is worth considering also with the key.

Reply:
knowing what key is potentially causing an issue could definitely be valuable during troubleshooting so I can see the benefit of leaving it here/enabled by default...

what about adding another attribute for the size of the payload being sent to memcached? that could also be valuable during troubleshooting

Reply:
Great suggestions! Pushed an update for addressing both of these. There's now a builder to configure if you want keys/values included in traces. Also added value size.